app/models/solution.rb

```diff
@@ -111,6 +111,7 @@ def update_latest_iteration_head_tests_status!(status)
 
     update_column(:latest_iteration_head_tests_status, status)
     Solution::SyncToSearchIndex.defer(self)
+    Solution::AutoUpdateToLatestExerciseVersion.(self)
```

Comment:
All the guarding is done in auto-update itself so there's none here. I considered deferring this, but as it's happening already in a background request and it's pretty lightweight, I decided against it.